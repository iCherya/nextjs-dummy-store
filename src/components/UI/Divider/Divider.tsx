type Props = {
  className?: string
}

export default function Divider({ className = '' }: Props) {
  return (
    <hr
      className={`w-48 h-1 mx-auto my-5 bg-gray-100 border-0 rounded ${className}`}
    />
  )
}
