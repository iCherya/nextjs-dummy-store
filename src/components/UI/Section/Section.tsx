import clsx from 'clsx'
import { ReactNode } from 'react'

import Divider from '@/components/UI/Divider/Divider'

type Props = {
  children: ReactNode
  className?: string
}

export default function Section({ children, className }: Props) {
  return (
    <>
      <section className={clsx('my-10', className)}>{children}</section>
      <Divider />
    </>
  )
}
