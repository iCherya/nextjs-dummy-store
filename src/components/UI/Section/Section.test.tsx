import { render, screen } from '@testing-library/react'

import Divider from '@/components/UI/Divider/Divider'
import Section from '@/components/UI/Section/Section'

jest.mock('@/components/UI/Divider/Divider', () => ({
  __esModule: true,
  default: jest.fn(() => 'DividerMock'),
}))

describe('Section', () => {
  it('should render children with Divider component', () => {
    const { container } = render(<Section>Children</Section>)

    expect(screen.getByText('Children')).toBeInTheDocument()

    expect(screen.getByText('DividerMock')).toBeInTheDocument()
    expect(Divider).toHaveBeenCalledTimes(1)

    expect(container).toMatchSnapshot()
  })
})
