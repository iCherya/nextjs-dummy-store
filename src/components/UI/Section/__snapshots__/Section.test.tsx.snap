// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section should render children with Divider component 1`] = `
<div>
  <section
    class="my-10"
  >
    Children
  </section>
  DividerMock
</div>
`;
